### ISSUE TITLE
Aggregate query causes skip/batch size controls to disappear and search results to be limited in 50 docs

### ISSUE THREAD

**Issue by LanceBeasley** [Studio3T/robomongo#1058]
Here is the query I was using:

```
db.getCollection('col_buckets').aggregate([
    {$match: {vDevId: "HMM1_5_F_17"}},
    {$limit: 500},
    {$sort: {startTime: -1}},
    {$unwind: "$values"},
    {$project: {'value': '$values.value', 'dischargeMs': '$values.dischargeMs', updateTime: "$values.updateTime",_id:0}},
    {$sort: {"updateTime": -1}}
    ])

```

Basically pulling values from an array in several documents.

The skip/batch controls disappear after running the query. The following screen shots show what happens.

Before running query...
![robomongoo before](https://cloud.githubusercontent.com/assets/13786778/13851964/2c9cc344-ec2e-11e5-92f4-e7e5426c1bb0.png)

After running query...
![robomongo after](https://cloud.githubusercontent.com/assets/13786778/13851968/300c7efc-ec2e-11e5-9404-2fe627cd9bec.png)

**Comment by kfitzgerald** [comment]
So it seems that 0.9 changed the way aggregates are run, using a cursor instead of just returning all documents. I'm reverting to the 0.8 releases until this can issue can be resolved. :(

**Comment by schetnikovich** [comment]
Confirmed. We will fix this problem in newest release. Thank you for reporting this issue!

**Comment by schetnikovich** [comment]
@LanceBeasley @kfitzgerald As a workaround, until we will fix this issue, you can view all documents of the `aggregate` command using the following script:

```
var cursor = db.collection.aggregate([ ... ]);
while (cursor.hasNext()) {
   print(cursor.next());
}
```

If you have less than 10,000 documents this will probably work for you. If you have more, or you still need to view documents in a smaller portions, you can implement simple paging yourself:

```
function showNextPage(cursor) {
  var n = 0;
  while (cursor.hasNext() && n < 50) {
    print(cursor.next());
    n++;
  }
}
```

And use it in the following way:

```
var cursor = db.collection.aggregate([ ... ]);
showNextPage(cursor);
```

If you want to view next page, select `nextPage(cursor)` text and run this selected script (`Ctrl/Cmd + Enter`):

![next-page](https://cloud.githubusercontent.com/assets/678660/14135355/afe8b866-f665-11e5-8478-ec0bf69b577d.png)

**Comment by LanceBeasley** [comment]
Thanks for the tip!


*Lance Beasley*
ENCO Electronic Systems <http://www.encoelectronics.com/>

**Comment by stasm** [comment]
As another workaround, you can specify `{ $out: "temp_result" }` as the last step of the aggregation pipeline which will put the results in the `temp_result` collection.  You can then view and query them as normal and drop the collection once you're done with it.

**Comment by WannerZoer** [comment]
any new report?

**Comment by juliashibalko** [comment]
Hi @WannerZoer, 
this issue is in our to-do list. Now we are working on Replica Set implementation. Please check our feature list for future release: https://github.com/paralect/robomongo#whats-planned-for-the-next-releases
At moment we include this ticket into release, we will get back to you with all updates!

**Comment by Nowaker** [comment]
Is there really no way to display more than 50 aggregated records?

**Comment by simsekgokhan** [comment]
Hi @Nowaker , currently the workaround below, from comment https://github.com/Studio3T/robomongo/issues/1058#issuecomment-203305989, seems to do the trick.

```
var cursor = db.collection.aggregate([ ... ]);
while (cursor.hasNext()) {
   print(cursor.next());
}
```

**Comment by abulhawa** [comment]
This does not work for me:
```
var cursor = db.collection.aggregate([ ... ]);
while (cursor.hasNext()) {
   print(cursor.next());
}
```
When I ran it, robomongo appeared busy for a few seconds then suddenly crashed. all unsaved scripts were lost.
env: windows 10 pro, Robomongo 1.0, MongoDB server 3.4.4

**Comment by simsekgokhan** [comment]
Hi @abulhawa , Robomongo 1.0 does not support MongoDB 3.4.
Please use [Robomongo 1.1 - Beta version with MongoDB 3.4 Support](https://github.com/Studio3T/robomongo/issues/1250#issuecomment-301036770).

**Comment by abulhawa** [comment]
Hi @simsekgokhan, thanks for this. I tried it, but Robomongo still crashes.

**Comment by simsekgokhan** [comment]
@abulhawa , I just tried with Windows 10, was not able to reproduce the crash.
Can you please create a new ticket here https://github.com/Studio3T/robomongo/issues/new.
Please also include crash report/log file  and exact/very similar script that you are running for further investigation.

**Comment by thechile** [comment]
still only shows 50 using latest 1.1 release.

**Comment by shishir99111** [comment]
**FINALLY GOT PERFECT SOLUTION AFTER TRYING FOR DAYS!!..**
try using `toArray` with the query, it will return all the records in Array.
eg:
`db.getCollection('debits').aggregate([{"$group" : { _id: "$timestamp"}}]).toArray()`

![screen shot 2017-06-21 at 1 42 34 pm](https://user-images.githubusercontent.com/13572684/27373907-9cdc545c-5687-11e7-8ae8-48be23cfa29f.png)

**Comment by Hades1996** [comment]
i'm using 1.1.1 version and still get this issue. 

".toArray()" trick works, is a good workaround.

**Comment by leandrosa** [comment]
#issuecomment-310002216 
You must be careful for use `.toArray()`, because it return all the documents to the memory. If many documents were returned, it can be fill the memory so much

**Comment by leandrosa** [comment]
You must be careful for use `.toArray()`, because it return all the documents to the memory. If many documents were returned, it can be fill the memory so much

**Comment by crertel-packlane** [comment]
It'd be great to have this fixed! What's the rough cause, if I can ask?

**Comment by donnyv** [comment]
Studio3T bought RoboMongo, thats why its never getting fixed.

**Comment by benevbright** [comment]
Thanks for the trick but why it has not fixed so long time.

**Comment by tomasznaumowicz** [comment]
As of the 0.9 release, results of a query are being returned as a cursor instead of just returning all documents. Otherwise one wouldn't be able to run larger queries. Not being able to iterate over the results makes this improvement insufficient. We will review this decision and -if needed - revert the change.

In the meantime, please use the `$out` stage to dump the results to a temp collection.

**Comment by abulhawa** [comment]
@RachelScodes it is working for me, robo 3T 1.2.1, on windows 10, MongoDB 3.6.3

**Comment by simsekgokhan** [comment]
Hi all, sorry for the long delay, this issue should be fixed in Robo 3T 1.2. 
Please see: https://blog.robomongo.org/robo-3t-1-2/#a2

@RachelScodes 
Your problem might be different than this ticket (which is for "Aggregate query results are not pagaeble" problem). Please also see the blog link above.
Can you give more details about your problem?  
Error screenshot, your script, versions of your OS, Robo, MongoDB. Thanks.
